When compiling with static linking, remove the "-rdynamic" option if present.
This resolves a problem with musl-cc where it would produce a binary dependent
on a non-existent musl shared object.

diff --git CMakeLists.txt CMakeLists.txt
index a25b269..4ef1a01 100755
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -26,6 +26,10 @@ option(CMARK_STATIC "Build static libcmark library" ON)
 option(CMARK_SHARED "Build shared libcmark library" ON)
 option(CMARK_LIB_FUZZER "Build libFuzzer fuzzing harness" OFF)
 
+if(CMARK_STATIC AND "${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+  string(REGEX REPLACE "( |^)-rdynamic( |$)" " " CMAKE_SHARED_LIBRARY_LINK_C_FLAGS ${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS})
+endif()
+
 add_subdirectory(src)
 if(CMARK_TESTS AND (CMARK_SHARED OR CMARK_STATIC))
   add_subdirectory(api_test)
